File: app/src/main/java/com/example/administrator/itemdecoration/callback/GroupImpl.java
package com.example.administrator.itemdecoration.callback;

import com.example.administrator.itemdecoration.bean.GroupInfo;

/**
 * Created by lengwei on 2017/11/11.
 *
 * @Description
 */
public class GroupImpl implements GroupCallback {

    @Override
    public GroupInfo group(int position) {
        int index = position % 5;
        int groupId = position / 5;
        GroupInfo info = new GroupInfo();
        info.setGroupSize(5);
        info.setGroupId(groupId + "");
        info.setPosition(index);
        info.setTitle(groupId + "");
        return info;
    }
}

File: app/src/main/java/com/example/administrator/itemdecoration/callback/GroupImpl.kt
package com.example.administrator.itemdecoration.callback

import com.example.administrator.itemdecoration.bean.GroupInfo

/**
 * Created by lengwei on 2017/11/11.
 *
 * @Description
 */
class GroupImpl : GroupCallback {

    override fun group(position: Int): GroupInfo {
        val index = position % 5
        val groupId = position / 5
        val info = GroupInfo()
        info.groupSize = 5
        info.groupId = groupId.toString() + ""
        info.position = index
        info.title = groupId.toString() + ""
        return info
    }
}
